contracts/reviewer.sol

```diff
@@ -60,62 +121,70 @@ contract ReviewerTransactionGuard is BaseGuard {
+        require(
+            getFunctionSelector(data) != removeGuardBytesData,
+            "This guard cannot be removed or changed!"
+        );
         // require(bytes4(data) != ENCODED_SIG_SET_GUARD, "This guard cannot be removed or changed!");
-        
+
         // Allows policy changes and rejections
-        if (to != safeAddress || to != address(this))
-        {
+        if (to != safeAddress || to != address(this)) {
             uint96 appId;
-            bytes32 applicantAddress;
+            address applicantAddress;
 
-            if(getFunctionSelector(data) == multiSendBytesData)
-            {
-                uint96 numTransfers = getApplicationId(data, 68)/217;
+            if (getFunctionSelector(data) == multiSendBytesData) {
+                uint96 numTransfers = getApplicationId(data, 68) / 217;
 
-                for(uint96 i = 0; i < numTransfers; i++)
-                {
-                    appId = getApplicationId(data, 32 + 221 + (i*217));
-                    applicantAddress = getPaymentAddress(data, 32 + 157 + (i*217));
+                for (uint96 i = 0; i < numTransfers; i++) {
+                    appId = getApplicationId(data, 32 + 221 + (i * 217));
+                    applicantAddress = getPaymentAddress(
+                        data,
+                        32 + 157 + (i * 217)
+                    );
 
                     fetchReviews(appId, applicantAddress);
                 }
-            }
-            
-            else 
-            {
+            } else {
                 appId = getApplicationId(data, 100);
-                applicantPaymentAddress = getPaymentAddress(data, 36);
-                
+                applicantAddress = getPaymentAddress(data, 36);
+
                 fetchReviews(appId, applicantAddress);
             }
         }
     }
 
-    function checkAfterExecution(bytes32 txHash, bool success) external override {
-    }
-
-    function fetchReviews(uint96 _appId, address _applicantPaymentAddress) public returns (uint96 k) {
+    function checkAfterExecution(bytes32 txHash, bool success)
+        external
+        override
+    {}
 
+    function fetchReviews(uint96 _appId, address _applicantPaymentAddress)
+        public view 
+    {
         address applicantWalletAddress;
-        (,,,applicantWalletAddress,,,,) = IApplicationRegistry(ApplicationRegistryAddress).applications(_appId);
-        address applicantZerowalletAddress = IApplicationRegistry(ApplicationRegistryAddress).walletAddressMapping(_applicantPaymentAddress);
-        
-        require(applicantZerowalletAddress == applicantPaymentAddress, "The proposal author, application and payment address have a mismatch");
+        (, , , applicantWalletAddress, , , , ) = applicationReg.applications(_appId);
+        address applicantZerowalletAddress = applicationReg.walletAddressMapping(_applicantPaymentAddress);
```

Comment:
The `_applicantPaymentAddress` needs to be converted to `bytes32` as the mapping requires a `bytes32` type as index.